extensions/wikia/AchievementsII/Ach_setup.php

```diff
@@ -118,36 +118,33 @@ function Ach_Setup() {
 	wfProfileOut(__METHOD__);
 }
 
-function Ach_GetMenu(&$nodes) {
+function Ach_GetMenu( array &$nodes ): bool {
 	$nodes[0]['children'][] = count($nodes);
-	$nodes[] = array(
-		//'original' => 'achievementsleaderboard',
+	$nodes[] = [
 		//the message is stored in /languages/messages/wikia/MessagesEn.php to avoid loading the i18n for the extension
-		'text' => wfMsg('achievements-leaderboard-navigation-item'),
-		'href' => Skin::makeSpecialUrl("Leaderboard"),
-  		//'depth' => 1,
-		//'parentIndex' => 0
-	);
+		'text' => wfMsg( 'achievements-leaderboard-navigation-item' ),
+		'href' => Skin::makeSpecialUrl( "Leaderboard" ),
+	];
 
 	return true;
 }
 
 function Ach_UploadVerification($destName, $tempPath, &$error) {
 	if (Ach_isBadgeImage($destName, true /* check user right to upload sponsored badge */ )) {
 
-		$error = wfMsgExt('achievements-upload-not-allowed', array('parse'));
+		$error = wfMessage('achievements-upload-not-allowed' )->parse();
 		return false;
 	}
 	return true;
 }
 
-function Ach_ArticleSaveComplete(	&$article, &$user, $text,
-					$summary, $minoredit, $watchthis,
-					$sectionanchor, &$flags, $revision,
-					&$status, $baseRevId) {
+function Ach_ArticleSaveComplete(
+	WikiPage $article, User $user, $text, $summary, $minoredit, $watchthis, $sectionanchor, $flags,
+	$revision, Status &$status, $baseRevId
+): bool {
 	wfProfileIn(__METHOD__);
 
-	if($status instanceof Status && $status->ok == true && count($status->errors) == 0) {
```

Comment:
In `WikiPage::doEdit` `$status` is created via `Status::newGood` call which always returns `Status` instance - and a property is accessed before it is passed on:

```		// Return the new revision (or null) to the caller
		$status->value['revision'] = $revision;

		Hooks::run( 'ArticleSaveComplete', [
			$this,
			$user,
			$text,
			$summary,
			$flags & EDIT_MINOR,
			null,
			null,
			$flags,
			$revision,
			&$status,
			$baseRevId,
		] );
```